ui/src/ScholarReader.tsx

```diff
@@ -313,6 +321,28 @@ class ScholarReader extends React.PureComponent<ScholarReaderProps, State> {
         if (symbols.length >= 1) {
           const mathMl = await api.mathMlForArxivId(this.props.paperId.id);
           this.setMathMl(mathMl);
+
+          /**
+           * Build a mapping from symbol to all possible parent matches
+           */
+          const symbolMatchSet: SymbolMatchSet = {};
+          symbols.map(s => {
+            symbolMatchSet[s.id] = new Set(
+              selectors.matchingSymbols(s, symbols, mathMl)
+              .map(m => {
+                // Need to get the top most parent as this is the 
+                // symbol that determines the annotation
```

Comment:
If you just get the parents, doesn't that only match each symbol to itself? If so, that's probably okay, though we wouldn't need to create a symbol match set (just do a literal comparison of the symbol within the shouldHighlight method).

Reply:
I am a little confused by what you mean by "only match each symbol to itself"? Since it's matching against selectors.matchingSymbols I believe it will match against every exact match but since the annotations are only at the top most parent, we need that id in order to do the actual highlighting I believe. However, I might be misunderstanding hahaha

Reply:
Sorry, it's my mistake. I think I get what's happening here!

First the call to `matchingSymbols()` gets all other symbols with a matching subtree:

```
Symbol (x_i) -> {All other symbols with matching a subtree like: x, i, x_i }
```

And then the subsequent `.map` gets the highest ancestor of those symbols, which will be the "clickable" symbol entity in the document:

```
{All symbols with (x, i, x_i) -> {All clickable symbols containing (x, i, x_i), including: e_x, x^2, ij}
```

Is that the way it works? If so, can we update the comment above with this clarification?

Reply:
Yes that is correct, I will update the comment :)!